services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -150,10 +150,7 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           ua.publish_date #{teacher_timezone_offset_string} AS publish_date,
           pre_activity.id AS pre_activity_id,
           cu.created_at AS unit_activity_created_at,
-          ps.release_method AS #{PackSequence::RELEASE_METHOD_KEY},
-          ps.id AS #{PackSequence::ID_KEY},
-          psi.id AS #{PackSequenceItem::ID_KEY},
-          psi.order AS #{PackSequenceItem::ORDER_KEY},
+          COALESCE(upsi.status, '#{UserPackSequenceItem::NO_LOCK}') AS user_pack_sequence_item_status,
```

Comment:
Ok, so `status` is a trinary, `LOCKED`, `UNLOCKED`, `NO_LOCK`. 

Take or leave, you might consider keeping this third state as `nil`, instead of renaming to `NO_LOCK`. It's not a big deal, but you are kinda committing to doing this `COALESCE` on every query for this attribute from this point on, which may be more of a gotcha than the logic having a relatively readable `status.nil?` check (There are pros to having the explicit `NO_LOCK` so this isn't a strongly held opinion, and it's always easy to change later).

Reply:
That makes sense.  I will just pass status instead and `nil` can be interpreted as `NO_LOCK` on the front-end